lib/utils/pmid.rb

```diff
@@ -20,89 +20,65 @@ def self.pubmed_url(pubmed_id)
 
   private
   def self.url_for_pubmed_id(pubmed_id)
-    "https://www.ncbi.nlm.nih.gov/pubmed/#{pubmed_id}?report=xml&format=text"
+    "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id=#{pubmed_id}&retmode=json&tool=DGIdb&email=help@dgidb.org"
   end
 
   class PubMedResponse
-    attr_reader :xml
-    def initialize(response_body)
-      @xml = Nokogiri::XML(Nokogiri::XML(response_body).text)
+    attr_reader :result
+    def initialize(response_body, pmid)
+      @result = JSON.parse(response_body)['result'][pmid]
     end
 
     def citation
       [first_author, year, article_title, journal].compact.join(', ')
     end
 
     def authors
-      xml.xpath('//AuthorList/Author').to_a.each.with_index(1).map do |author, i|
-        {
-          fore_name: author.xpath('ForeName').text,
-          last_name: author.xpath('LastName').text,
-          author_position: i
-        }
-      end
+      result['authors'].map{|a| a['name']}
     end
 
     def pmc_id
-      xpath_contents_or_nil("//PubmedData/ArticleIdList/ArticleId[@IdType='pmc']")
+      result['articleids'].each do |articles|
+        if articles['idtype'] == 'pmcid'
+          return articles['value']
+        end
+      end
     end
 
-    def abstract
-      xpath_contents_or_nil('//Abstract/AbstractText')
-    end
+    #def abstract
+    #  xpath_contents_or_nil('//Abstract/AbstractText')
+    #end
 
     def first_author
-      xpath_contents_or_nil('//AuthorList/Author[1]/LastName') do |author_name|
-        if xml.xpath('//AuthorList/Author').size > 1
-          author_name + " et al."
-        else
-          author_name
-        end
+      if authors.size > 1
+        authors.first + " et al."
+      else
+        authors.first
       end
     end
 
     def publication_date
-     [day, month, year]
+      result['pubdate']
     end
 
     def year
-      xpath_contents_or_nil('//Journal/JournalIssue/PubDate/Year')
-    end
-
-    def month
-      monthname = xpath_contents_or_nil('//Journal/JournalIssue/PubDate/Month')
-      if monthname
-        Date::ABBR_MONTHNAMES.index(monthname)
+      if Date.new(publication_date.to_i).gregorian?
+        publication_date
```

Comment:
this should just be publication_date.to_i if we want to extract only the year, but won't all our years be gregorian?

https://ruby-doc.org/stdlib-2.4.1/libdoc/date/rdoc/Date.html#method-i-gregorian

Reply:
This doesn't quite seem to do what I wanted it to. Sometimes the date in this field is just the year and this if-clause was supposed to catch that. I will rework this.